library/pkparse.c

```diff
@@ -1343,6 +1343,7 @@ int mbedtls_pk_parse_key( mbedtls_pk_context *pk,
      * error
      */
 #if defined(MBEDTLS_PKCS12_C) || defined(MBEDTLS_PKCS5_C)
+    if( pwdlen != 0 )
```

Comment:
This relies on ret being set (when arriving at line 1362), which was previously done within the block that follows (at line 1355), but now relies on it being set before it gets here. It is in fact always set some distance above, so the logic works, but it's a little less obviously correct. Possibly it is clearer to add an else clause at line 1360 which sets e.g. `ret = MBEDTLS_ERR_PEM_NO_HEADER_FOOTER_PRESENT`. But I'm ok with this either way.